### PR TITLE
[WIP] Zero initialize device_uvector before scatter

### PR DESCRIPTION
**Pull request by zhangjianting**
bug fix based on [discussions](https://github.com/rapidsai/cuspatial/issues/318)

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: cpp/src/indexing/construction/detail/phase_2.cuh [hunk 1]
```diff
@@ -193,6 +193,8 @@ inline rmm::device_uvector<uint32_t> compute_parent_positions(
                            position_map.begin());
     // line 2 of algorithm in Fig. 5 in ref.
     rmm::device_uvector<uint32_t> parent_pos(num_child_nodes, stream);
+    thrust::uninitialized_fill(rmm::exec_policy(stream)->on(stream),
```
Comment (harrism) [inline cpp/src/indexing/construction/detail/phase_2.cuh:196]: Run clang-format on your code before submitting.  

```
python cpp/scripts/run-clang-format.py
```

**Review by trxcllnt — APPROVED** [review]

File: cpp/src/indexing/construction/detail/phase_2.cuh (hunk 1 above)
Comment (trxcllnt) [reply cpp/src/indexing/construction/detail/phase_2.cuh]: I fixed the lint and merged the build fixes in.

Comment (zhangjianting) [reply cpp/src/indexing/construction/detail/phase_2.cuh]: Thanks Paul. Having been busy in the past few days before coming back.

**Comment by trxcllnt** [comment]
rerun tests

**Review by harrism — APPROVED** [review]

**Comment by drabastomek** [comment]
Confirming this is resolved in 0.17

**Pull request merged**